include/types.hpp

```diff
@@ -92,5 +94,23 @@ using HWVerList = std::vector<std::pair<std::string, std::string>>;
 using SystemTypeMap =
     std::unordered_map<std::string, std::pair<std::string, HWVerList>>;
 
+using WriteOnDbusIpzFormat = std::tuple<sdbusplus::message::object_path, std::string, std::string, BinaryVector>;
+using WriteOnDbusKwFormat = std::tuple<sdbusplus::message::object_path, std::string, BinaryVector>;
+using WriteOnHwIpzFormat = std::tuple<std::string, std::string, std::string, BinaryVector>;
+using WriteOnHwKwFormat = std::tuple<std::string, std::string, BinaryVector>;
+
+using WriteKeywordSignature = std::variant<WriteOnDbusIpzFormat, WriteOnDbusKwFormat, WriteOnHwIpzFormat, WriteOnHwKwFormat>;
+
+using ReadFromDbusIpzFormat = std::tuple<sdbusplus::message::object_path, std::string, std::string>;
+using ReadFromDbusKwFormat = std::tuple<sdbusplus::message::object_path, std::string>;
+using ReadFromHwIpzFormat = std::tuple<std::string, std::string, std::string>;
+using ReadFromHwKwFormat = std::tuple<std::string, std::string>;
+
+using ReadKeywordSignature = std::variant<ReadFromDbusIpzFormat, ReadFromDbusKwFormat, ReadFromHwIpzFormat, ReadFromHwKwFormat>;
+
+using ReturnIpzData = std::tuple<std::string, std::string, BinaryVector>;
+using ReturnKwData = std::tuple<std::string, BinaryVector>;
+
+using ReturnData = std::variant<ReturnIpzData, ReturnKwData>;
 } // namespace types
-} // namespace vpd
\ No newline at end of file
+} // namespace vpd
```

Comment:
input to read : objpath, record, keyword
output after reading : record, keyword, value. 

data type won't be same for both input and output on reads.

input to write : objpath, record, keyword, value
output after writing : record, keyword, value ( as per our discussion, no need to return back object path. )

here too, data type can't be same for both input and output.

Reply:
i found that, with your suggestion we end up changing the structure by having two tuples. the type should just be std::tuple<string, string, string, vector<uint8_t>>. and not
std::tuple<string, std::tuple<string, string, vector<uint8_t>

as per your concern on variable reusability, i have modified the variable names accordingly. please take a look

Reply:
why we need nested tuples here? just because we achieve variable reusability?
using WriteVpdType = std::variant<std::tuple<std::string, IpzDataFormat>, std::tuple<std::string, kwdDataFormat>>;